webview-ui/src/components/ResolutionsPage/ResolutionsPage.tsx

```diff
@@ -60,94 +49,69 @@ const ResolutionPage: React.FC = () => {
-
-            {isFetchingSolution && <Spinner />}
-
-            {hasResponse && (
-              <ReceivedMessage>
-                <FileChanges
-                  changes={getRemainingFiles()}
-                  onFileClick={handleFileClick}
-                  onApplyFix={handleAcceptClick}
-                  onRejectChanges={handleRejectClick}
+      <PageSection isWidthLimited>
+        <Chatbot displayMode={ChatbotDisplayMode.embedded}>
+          <ChatbotContent>
+            <MessageBox>
+              {isTriggeredByUser && (
+                <>
+                  <SentMessage mainContent="Here is the scope of what I would like you to fix:">
+                    <ChatCard color="yellow">
+                      <IncidentTableGroup
+                        onIncidentSelect={handleIncidentClick}
+                        incidents={solutionScope.incidents}
+                      />
+                    </ChatCard>
+                  </SentMessage>
+                  <SentMessage mainContent="Please provide resolution for this issue."></SentMessage>
+                </>
+              )}
+              {hasNothingToView && <ReceivedMessage mainContent="No resolutions available." />}
+              {isHistorySolution && <ReceivedMessage mainContent="Loaded last known resolution." />}
+              {chatMessages.map((msg) => (
+                <ReceivedMessage
+                  timestamp={msg.timestamp}
+                  key={msg.value.message as string}
+                  mainContent={msg.value.message as string}
                 />
-              </ReceivedMessage>
-            )}
-            {hasEmptyResponse && !hasResponseWithErrors && (
-              <ReceivedMessage>Received response contains no resolutions.</ReceivedMessage>
-            )}
-            {hasResponseWithErrors && (
-              <>
-                <ReceivedMessage>Response contains errors:</ReceivedMessage>
+              ))}
+              {isFetchingSolution && <ReceivedMessage isLoading />}
+              {hasResponse && (
                 <ReceivedMessage>
-                  <ul>
-                    {resolution.encountered_errors.map((error, index) => (
-                      <li key={index}>{error}</li>
-                    ))}
-                  </ul>
+                  <FileChanges
+                    changes={getRemainingFiles()}
+                    onFileClick={handleFileClick}
+                    onApplyFix={handleAcceptClick}
+                    onRejectChanges={handleRejectClick}
+                  />
                 </ReceivedMessage>
-              </>
-            )}
-
-            {isResolved && !isFetchingSolution && (
-              <ReceivedMessage>All resolutions have been applied.</ReceivedMessage>
-            )}
-          </Flex>
-        </Flex>
+              )}
+              {hasEmptyResponse && !hasResponseWithErrors && (
+                <ReceivedMessage>Received response contains no resolutions.</ReceivedMessage>
```

Comment:
What is the difference between the text being `children` vs `mainContent`?  I'd prefer using one or the other to be consistent if there is no real difference.

Reply:
Updated the props to just use "content" for string content and "extraContent" for React elements.